examples/text-generation/textgen_pipeline/pipeline.py

```diff
@@ -0,0 +1,219 @@
+            if load_to_meta:
+                ds_inference_kwargs["checkpoint"] = checkpoints_json
+
+            model = deepspeed.init_inference(model, **ds_inference_kwargs)
+            model = model.module
+        else:
+            get_repo_root(model_name_or_path)
+            model = AutoModelForCausalLM.from_pretrained(model_name_or_path, torch_dtype=model_dtype)
+            model = model.eval().to(self.device)
+            is_optimized = model_is_optimized(model.config)
+            model = wrap_in_hpu_graph(model)
+
+        self.model = model
+
+        # Used for padding input to fixed length
+        self.tokenizer.padding_side = "left"
+        self.max_padding_length = kwargs.get("max_padding_length", self.model.config.max_position_embeddings)
+
+        # Define config params for llama models
+        if self.model.config.model_type == "llama":
+            self.model.generation_config.pad_token_id = 0
+            self.model.generation_config.bos_token_id = 1
+            self.model.generation_config.eos_token_id = 2
+            self.tokenizer.bos_token_id = self.model.generation_config.bos_token_id
+            self.tokenizer.eos_token_id = self.model.generation_config.eos_token_id
+            self.tokenizer.pad_token_id = self.model.generation_config.pad_token_id
+            self.tokenizer.pad_token = self.tokenizer.decode(self.tokenizer.pad_token_id)
+            self.tokenizer.eos_token = self.tokenizer.decode(self.tokenizer.eos_token_id)
+            self.tokenizer.bos_token = self.tokenizer.decode(self.tokenizer.bos_token_id)
+
+        # Applicable to models that do not have pad tokens
+        if self.tokenizer.pad_token is None:
+            self.tokenizer.pad_token = self.tokenizer.eos_token
+            self.model.generation_config.pad_token_id = self.model.generation_config.eos_token_id
+
+        # Edit generation configuration based on input arguments
+        self.generation_config = copy.deepcopy(self.model.generation_config)
+        self.generation_config.max_new_tokens = kwargs.get("max_new_tokens", 100)
+        self.generation_config.use_cache = kwargs.get("use_kv_cache", True)
+        self.generation_config.static_shapes = is_optimized
+        self.generation_config.do_sample = kwargs.get("do_sample", False)
+        self.generation_config.num_beams = kwargs.get("num_beams", 1)
+        self.generation_config.temperature = kwargs.get("temperature", 1.0)
+        self.generation_config.top_p = kwargs.get("top_p", 1.0)
+        self.generation_config.repetition_penalty = kwargs.get("repetition_penalty", 1.0)
+        self.generation_config.num_return_sequences = kwargs.get("num_return_sequences", 1)
+        self.generation_config.bad_words_ids = None
+        self.generation_config.force_words_ids = None
+        self.generation_config.ignore_eos = False
+
+        if self.use_deepspeed:
+            torch.distributed.barrier()
```

Comment:
Since `run_generation.py` was recently refactored, let's try to use here the methods defined in https://github.com/huggingface/optimum-habana/blob/main/examples/text-generation/utils.py.
That way we avoid duplicated code.